### ISSUE TITLE
[CI/CD]: Review coverage of accessibility checks in EDU feedback-tool end-to-end tests

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#7349]
**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

Applications **must** have thorough end-to-end tests that run in our continuous integration/continuous deployment (CI/CD) pipeline. These tests should include thorough axe checks. While auditing the `/edu-benefits/feedback-tool` folder, I noticed we could be opening one or two accordions on the review form page before running the axe check. We should also remove the yellow focus halo from the `<h1>` on the confirmation page. I'd like the front-end engineering team to review the applications, and update the tests to review this hidden content. Definition of done in acceptance criteria below.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Environment
* `vets-website/src/applications/edu-benefits/feedback-tool/tests`
* `$ yarn test:e2e src/applications/edu-benefits/feedback-tool/tests`

## Acceptance Criteria

**Definition of done:**
- [ ] Front-end team member(s) have reviewed end-to-end tests for axe checks
- [ ] Open 1 or 2 accordions on the review page (last page before submitting) and run the axe check
- [ ] Remove the yellow focus halo from the `<h1>` on the confirmation page. We're leaning toward showing the halo only on focusable elements.
- [ ] FE team has consulted with accessibility specialist in cases where there are high numbers of modals, accordions, other hidden content that could slow down e2e test runs.
- [ ] No axe `violations` appear in the plugin console. These will break the CI/CD build.

## WCAG or Vendor Guidance (optional)

* [Custom axeCheck helper method](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/testing/e2e/nightwatch-commands/axeCheck.js)
* [VSP guidance on writing end-to-end tests](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/platform/quality-assurance/e2e-testing)

**Comment by 1Copenut** [comment]
@mattpointzxer0 If the team has moved to use the Cypress forms testing, would you mind commenting here and at include me? The Cypress forms tester has axe checks built in and would be a good way to close this issue without additional work.

**Comment by 1Copenut** [comment]
@mattpointzxer0 I'd like to revisit this issue when things are settled a bit with COVID related work. I'm taking another look at the e2e test coverage for accessibility checks with recent work to make the Cypress axeCheck helper more robust and easier to modify. Thank you!

**Comment by joshkimux** [comment]
@Mottie another one of these tickets. Two questions:
- Is there a way I can check if this is something that can be closed so I don't need to bug you?
- (if not) Is this something that can be closed? (ONLY do this if you've run out of things to do and it's still the work day, there's no deadline for this, and I can also reach out to CAIA to look into this ❤️ )

**Comment by Mottie** [comment]
Hmm, I can't think of an easy way to determine if something has already been done without actually looking at the code; but in this case, it depends:

- When using the form tester, Cypress tests should _all_ be using common code to expand all accordions and additional info containers, then run the axe core check. This includes the review & submit page.
- For custom apps and custom pages within a form, I can't say 100% for sure that all forms are following this convention, but there is an ESLint rule that underlines the tests in the code editor to let developers know that an axe check is missing. They _would_ have to include a Cypress command to `expandAccordions` on pages with an accordion and/or additional info components.

That said... in the `edu-benefits/tests/feedback-tool` this is what I see:

- `00.feedback-tool.cypress.spec.js` - looks good
- `01.feedback-tool-keyboard.cypress.spec.js` - no axe checks (ESLint messaging), but appears to follow the same path as the first test; and, it should be using keyboard only commands, but the form start button uses a mouse click
- `02.feedback-tool-keyboard-navigation.cypress.spec.js` - starts with an axe check on the first page, uses keyboard navigation throughout, but doesn't run an axe check on any other page

If we consider that the first test covers all the pages, I don't think we need axe checks in the other two. So I think we're okay to close this ticket.
